framework/src/packets/icmp/v6/ndp/options/link_layer_addr.rs

```diff
@@ -30,12 +31,36 @@ use std::fmt;
 
 #[derive(Debug)]
 #[repr(C, packed)]
-struct LinkLayerAddressFields {
+pub struct LinkLayerAddressFields {
     option_type: u8,
     length: u8,
     addr: MacAddr,
 }
 
+impl Default for LinkLayerAddressFields {
+    fn default() -> LinkLayerAddressFields {
+        LinkLayerAddressFields {
+            option_type: 0,
```

Comment:
the option_type will either be 1 or 2 depending on which option type you want. that will be set in router/test_utils. we could default this to 1 if we wanted to and only have the test_utils set it to 2 if we need the target layer address option

Reply:
maybe it's which is more common, but that may be lame criteria.

Reply:
i'm not sure which is more common TBH since we havent done any work on router solicitations yet. router adverts will only have source LL addresses and not targets

Reply:
ah...this isnt going to work since it will change the signature of the method. option_type = 0 isn't really an option type at all since options start at 1. we can default it to 1(source LL) or 2(target LL) . we should default it to whatever the most common of the message types(router advertisement or router solicits) but TBH, i have no idea which we will see more of

Reply:
i tried separating them out and it was a lot of duped code. since we are passing a pointer for the fields, we cant default the value in a constructor. you get a pointer cant be deref'd. since you cant declare the default value, you would still have to do `set_option_type` each time you create the link layer address option. i dont think its worth it to separate it out IMO

Reply:
yep. i added a check to make sure a valid option was passed in